packages/flutter/lib/src/material/button_theme.dart

```diff
@@ -284,6 +314,197 @@ class ButtonThemeData extends Diagnosticable {
   /// This property only affects [DropdownButton] and its menu.
   final bool alignedDropdown;
 
+  final Color buttonColor;
+  final Color disabledColor;
+  final Color highlightColor;
+  final Color splashColor;
+  final ColorScheme colorScheme;
+  final MaterialTapTargetSize materialTapTargetSize;
+
+  Brightness getBrightness(MaterialButton button) {
+    return button.colorBrightness ?? colorScheme.brightness;
+  }
+
+  ButtonTextTheme getTextTheme(MaterialButton button) {
+    return button.textTheme ?? textTheme;
+  }
+
+  bool _isFlatButton(MaterialButton button) {
+    return button.type == 'FlatButton' || button.type == 'FlatButton.icon';
+  }
+
+  bool _isRaisedButton(MaterialButton button) {
+    return button.type == 'RaisedButton' || button.type == 'RaisedButton.icon';
+  }
+
+  bool _isOutlineButton(MaterialButton button) {
+    return button.type == 'OutlineButton' || button.type == 'OutlineButton.icon';
+  }
+
+  bool _isIconButton(MaterialButton button) {
+    return button.type == 'RaisedButton.icon' || button.type == 'FlatButton.icon' || button.type == 'OutlineButton.icon';
+  }
+
+  Color _getDisabledColor(MaterialButton button) {
+    return getBrightness(button) == Brightness.dark
+      ? colorScheme.onSurface.withOpacity(0.30)  // default == Colors.white30
+      : colorScheme.onSurface.withOpacity(0.38); // default == Colors.black38;
+  }
+
+  Color getDisabledTextColor(MaterialButton button) {
```

Comment:
Each of these properties should probably have extensive documentation on how they are computed - possibly for each button type.